src/core/currency/wallet/currency-wallet-pixie.ts

```diff
@@ -233,6 +234,10 @@ export const walletPixie: TamePixie<CurrencyWalletProps> = combinePixies({
             // Turn synchronous errors into promise rejections:
             startupPromise = Promise.resolve()
               .then(() => engine.startEngine())
+              .then(() => {
+                // Signal that engine startup is complete:
+                input.onOutput({ ...walletOutput, engineStarted: true })
+              })
```

Comment:
I'm pretty sure `combinePixies` takes care of the merging, and we only need to do `input.onOutput(true)` here.

Since you've put this extra `{}` in here, I'm guessing your debugger will show something like:

```js
{
  walletApi: {...},
  engine: {...},
  engineStarted: {
    walletApi: {...},
    engine: {...},
    engineStarted: true
  }
}
```

Can we verify that it's messed up in this way, and then fix it?

Reply:
Good catch.